docs/threading.rst

```diff
@@ -7,15 +7,39 @@ Threading model
 
    import rticonnextdds_connector as rti
 
-RTI Connector only allows the concurrent use of different :class:`Connector`
-instances. Method calls to the same :class:`Connector` instance or any of its contained
-entities are not thread safe, and the application is responsible to protect them
-if they're used concurrently.
+Operations on the same :class:`Connector` instance or any contained :class:`Input`,
+:class:`Output` are in general not protected for multi-threaded access. The only
+exceptions are the following *wait* operations.
+
+Thread-safe operations:
+   * :meth:`Connector.wait` (wait for data on any ``Input``)
+   * :meth:`Output.wait` (wait for acknowledgments)
+   * :meth:`Output.wait_for_subscriptions`
+   * :meth:`Input.wait` (wait for data on this ``Input``)
+   * :meth:`Input.wait_for_publications`
+
+These operations can block a thread while the same ``Connector`` is used in
+a different thread.
+
+.. note::
+
+   Currently :meth:`Input.wait` and :meth:`Input.wait_for_publications` cannot
+   be both called at the same time on the same ``Input`` instance.
+
+.. note::
+
+   :meth:`Output.write` can block the current thread under certain
+   circumstances, but :meth:`Output.write` is not thread-safe.
+
+All operations on **different** :class:`Connector` instances are thread-safe.
```

Comment:
Do you mean to say that calling set_max_objects_per_thread is thread-safe? It is, but since you have to call it at the beginning of your app before you even create any Connector, I don't see any advantage to mention it here.

Reply:
oh, I thought it wasn't thread safe, and if you did it in 2 Connector (defined in the same XML) you could have issues (since get qos -> set qos isn't atomic)

Reply:
if you call it from two different threads with different values, it will just end up configured with one of them randomly. But that's "safe."